src/lib/processors/watch-processor.ts

```diff
@@ -235,7 +239,10 @@ export function registerWatchEventHandlers(
     [WatchEvent.GIVE_UP]: err => {
       // If failure continues, log and exit
       logEvent(WatchEvent.GIVE_UP, err.message);
-      process.exit(1);
+      throw new Error(
+        "WatchEvent GiveUp Error: The watch has failed to start after several attempts.",
```

Comment:
Open to suggestions on what the error message should say.

Reply:
The KFC will emit this message just for transparency, it will be logged on L241 so I think we are good

```bash
this.#events.emit(
          WatchEvent.GIVE_UP,
          new Error(`Retry limit (${this.#watchCfg.resyncFailureMax}) exceeded, giving up`),
        );
```